server/session.go

```diff
@@ -130,7 +130,7 @@ func (p *Plugin) addUserSession(state *channelState, userID, connID, channelID s
 		}
 	}
 
-	if state.Call.HostID == "" && userID != p.getBotID() {
+	if state.Call.HostID == "" && userID != p.getBotID() && !state.Call.HostLocked {
 		state.Call.HostID = userID
 	}
```

Comment:
Slightly unrelated but maybe it would be cleaner if we always call `state.Call.getHostID()` here and keep any specific logic inside the method. Same would apply for when removing a session.

Reply:
That's nicer, keeps the logic all in one place, thanks.